### DISCUSSION TITLE
refetchInterval doesn't work with multiple useQuery usages

### DISCUSSION THREAD

**Discussion by pierissimo** [TanStack/query#417] [General]
If my component uses multiple useQuery hooks, and each hook have refetchInterval configured, the interval doesn't seem to fire for all the 3 queries, but only for the first one.

Here the reproduced bug (open the console): https://codesandbox.io/s/exciting-fire-tyypj?file=/src/App.js

**Comment by tannerlinsley** [comment]
As of today, it will take the shortest interval and only run that one interval as a least common denominator. Work is being done to support mixed intervals.

**Comment by tannerlinsley** [comment]
I'll consider this a feature request and move it to discussions.

**Comment by cherniavskii** [comment]
That's interesting, looks like a bug to me.
@tannerlinsley note, that each query uses different query key, so refetch interval should work for each of them.
I think it might be solved by https://github.com/tannerlinsley/react-query/pull/404, let's see

**Reply by tannerlinsley** [reply]
Each query key should already work separately, but instances no. Right now (or after your PR), it should use the shortest interval length per query, right?

**Reply by cherniavskii** [reply]
> Each query key should already work separately

yes, it should, but it doesn't because of https://github.com/tannerlinsley/react-query/discussions/399

I've checked @pierissimo 's example on latest master branch and it works as expected now :tada:

**Reply by cherniavskii** [reply]
This discussion should also be converted into issue

**Reply by tannerlinsley** [reply]
Yeah. Unfortunately there’s no button to do that yet.

**Reply by cherniavskii** [reply]
Oh, that's unfortunate

**Comment by cherniavskii** [comment]
Works as expected with 1.3.0 - https://codesandbox.io/s/busy-fast-nv7sx?file=/src/App.js

Note, that I've changed returning value to string instead of date in query functions, since react-query doesn't compare dates properly (data comparison is a new feature in 1.3.0)

**Reply by pierissimo** [reply]
It works great, even with different refetch intervals for each one.
Thanks!

**Reply by cherniavskii** [reply]
You're welcome!
